### ISSUE TITLE
Feature pipeline correlation filter causes train/inference dimension mismatch

### ISSUE THREAD

**Issue by Pr0mp7** [Photonensauger/Quant-2.0#5]
## Problem

The `FeaturePipeline.fit_transform()` correlation filter (`_correlation_filter`) drops different features depending on the input data range. This causes a **dimension mismatch** between training and inference (backtest), making the models unable to produce predictions.

### Reproduction

```bash
# 1. Train on 10 years of data
python scripts/train_model.py --model all --assets GLD --interval 1d --epochs 50 --days 3650
# -> Correlation filter drops 12 features -> 41 features -> models saved with input_dim=41

# 2. Backtest on 2 years (out-of-sample)
python scripts/run_backtest.py --strategy ensemble --assets GLD --interval 1d --start 2024-02-01 --end 2026-02-17
# -> Correlation filter drops 14 features -> 39 features -> model expects 41 -> crash
```

### Error

```
Feature/model error for GLD at bar 496: linear(): input and weight.T shapes cannot be multiplied (60x39 and 41x64)
```

This repeats for every bar. The backtest completes with **0 trades** and flat equity because no signal is ever generated.

### Root Cause

`FeaturePipeline._correlation_filter()` computes pairwise correlations on the input data and drops features above a threshold (0.95). Different data ranges have different correlation structures, so:

- **10yr data** → drops 12 features → **41 features** (models trained with this)
- **2yr data** → drops 14 features → **39 features** (backtest uses this)

The filter is stateless — it doesn't remember which columns it dropped during training.

### Current Workaround

Run the backtest with the **same date range** as training so the correlation filter produces the same features:

```bash
python scripts/run_backtest.py --strategy ensemble --assets GLD --interval 1d --start 2016-02-01 --end 2026-02-17
# -> 41 features -> matches model -> 180 trades
```

This works but defeats the purpose of out-of-sample testing.

---

### Option A: Save and reload the column mask (recommended)

Make the feature pipeline persist its state after `fit_transform()` and reuse it during inference.

1. After `_correlation_filter()`, save the list of **kept columns** to the pipeline state
2. Add a `transform()` method that applies the saved mask without re-computing correlations
3. Save the column mask alongside model checkpoints (or in a separate `.json`)
4. During backtest, load the mask and use `transform()` instead of `fit_transform()`

**Pros:** Clean separation of fit vs transform, standard ML pattern, no data range dependency
**Cons:** Need to manage an extra artifact (column mask file) alongside checkpoints

### Option B: Disable correlation filter during inference

Skip `_correlation_filter()` entirely during backtest/live and keep all features. Models would need to be trained on the full (unfiltered) feature set.

**Pros:** Simple, no state management
**Cons:** More features = more noise, models may overfit, larger input dimension

### Option C: Fix the feature set statically

Hardcode which features to keep (remove the dynamic correlation filter). Define a fixed feature list in config.

**Pros:** Fully deterministic, no mismatch possible
**Cons:** Loses adaptability, may keep redundant features for some assets

---

## Recommendation

**Option A** is the standard approach (sklearn's `fit()` + `transform()` pattern). The correlation filter should behave like a fitted transformer — learn the mask on training data, then apply it consistently during inference.

## Affected Files

- `quant/features/pipeline.py` — `FeaturePipeline._correlation_filter()` and `fit_transform()`
- `quant/backtest/engine.py` — `_generate_signal()` calls `fit_transform()` instead of `transform()`
- `quant/models/trainer.py` — should save feature metadata with checkpoints
